src/asmcnc/skavaUI/widget_status_bar.py

```diff
@@ -151,18 +154,38 @@ def __init__(self, **kwargs):
         self.sm=kwargs['screen_manager']
         Clock.schedule_interval(self.refresh_grbl_label_values, self.GRBL_REPORT_INTERVAL)      # Poll for status
         Clock.schedule_interval(self.refresh_ip_label_value, self.IP_REPORT_INTERVAL)      # Poll for status
+        #Clock.schedule_interval(self.check_limit_switch, self.IP_REPORT_INTERVAL)
 
     def on_enter(self):
         self.refresh_ip_label_value()
 
+    def check_limit_switch(self, dt):
+        if self.m.s.limit_x:
+            self.grbl_xm_label.text = 'm[color=ff0000]x[/color]:\n' + str(round(self.m.mpos_x(), 2))
+        elif self.m.s.limit_X:
+            self.grbl_xm_label.text = 'm[color=ff0000]X[/color]:\n' + str(round(self.m.mpos_x(), 2))
+        else:
+            self.grbl_xm_label.text = 'mX:\n' + str(round(self.m.mpos_x(), 2))
+
+        if self.m.s.limit_Y_axis:
+            self.grbl_ym_label.text = 'm[color=ff0000]Y[/color]:\n' + str(round(self.m.mpos_y(), 2))
+        elif self.m.s.limit_Y:
+            self.grbl_ym_label.text = 'm[color=ff0000]Y[/color]:\n' + str(round(self.m.mpos_y(), 2))
+        elif self.m.s.limit_y:
+            self.grbl_ym_label.text = 'm[color=ff0000]y[/color]:\n' + str(round(self.m.mpos_y(), 2))
+        else:
+            self.grbl_ym_label.text = 'mY:\n' + str(round(self.m.mpos_y(), 2))
+
+        if self.m.s.limit_z:
+            self.grbl_zm_label.text = 'm[color=ff0000]Z[/color]:\n' + str(round(self.m.mpos_z(), 2))
+        else:
+            self.grbl_zm_label.text = 'mZ:\n' + str(round(self.m.mpos_z(), 2))

```

Comment:
I think this function can be tidied up so that it's a bit more readable, and also testable. At the moment there's a lot of very similar statements. I can't see an obvious way to do it better, but my brain is also not braining well at the moment, so maybe just have a think about it :). It looks completely functional though :)